### PR TITLE
Set user when queueing VM actions

### PR DESCRIPTION
**Pull request by jntullo**
When VM actions are queued, the user should be set so that it is available as the requester in automate.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1546375

Please review @lfu

### CODE REVIEW

**Comment by jntullo** [comment]
The code climate issues mentioned are not something that should be addressed here

File: app/controllers/api/vms_controller.rb [hunk 1]
```diff
@@ -477,5 +478,18 @@ def request_console_vm(vm, protocol)
     rescue => err
       action_result(false, err.to_s)
     end
+
+    def queue_options(method)
+      current_user = User.current_user
```
Comment (lfu) [inline app/controllers/api/vms_controller.rb]: Does current user always exists for API requests?

Comment (abellotti) [reply app/controllers/api/vms_controller.rb]: yes, we set that based on the authentication (basic auth/auth/system).

**Comment by lfu** [comment]
@jntullo These changes don't cover VM retirement.

File: app/controllers/api/vms_controller.rb [hunk 2]
```diff
@@ -322,55 +323,55 @@ def validate_vm_for_remote_console(vm, protocol = nil)
...
-      task_id = queue_object_action(vm, desc, :method_name => "shelve_offload", :role => "ems_operations")
+      task_id = queue_object_action(vm, desc, queue_options("shelve_offload"))
       action_result(true, desc, :task_id => task_id)
     rescue => err
       action_result(false, err.to_s)
     end
 
     def destroy_vm(vm)
       desc = "#{vm_ident(vm)} deleting"
-      task_id = queue_object_action(vm, desc, :method_name => "destroy")
```
Comment (abellotti) [inline app/controllers/api/vms_controller.rb]: hmm, was the :role missing here by mistake ? we're now introducing it with queue_options.

**Comment by abellotti** [comment]
don't we need the same done for the instances controller actions ?

**Comment by jntullo** [comment]
@lfu currently retirement is not queuing an action, but instead calling [this method](https://github.com/ManageIQ/manageiq/blob/9291810e126cd34e5931bbf985be430a7018a422/app/models/mixins/retirement_mixin.rb#L65-L96)...does this still need the user? It does not seem like it would be used

**Comment by lfu** [comment]
@jntullo I made a PR https://github.com/ManageIQ/manageiq/pull/17033 to fix the retirement issue.

**Comment by jntullo** [comment]
@abellotti added it for instance actions as well

**Comment by lfu** [comment]
It works good, even for retirement with these 2 PRs.

**Comment by abellotti** [comment]
Thanks @jntullo for fixing this. LGTM!! 👍

**Comment by simaishi** [comment]
Fine backport (to manageiq repo) details:

```
$ git log -1
commit c8aec62ba354b218bc75e0f6a88ba54dc19420f5
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Wed Feb 21 11:13:56 2018 -0500

    Merge pull request #326 from jntullo/bz_1546375
    
    Set user when queueing VM actions
    (cherry picked from commit c6e8b4f9507292bf98032a107d67a55613b50243)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1565262
```

**Comment by simaishi** [comment]
This PR can't go to Fine branch as is, and caused Travis failures... after talking with @abellotti, reverted the backport.

```
commit cb5816dfaa5d52288da60fc638e87d3cfd188514
Author: Satoe Imaishi <simaishi@redhat.com>
Date:   Wed Apr 11 12:07:49 2018 -0400

    Revert "Merge pull request #326 from jntullo/bz_1546375"
    
    This reverts commit c8aec62ba354b218bc75e0f6a88ba54dc19420f5.
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1565262
```

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 6b3e36c48702c48158b1aec8e86534c001bead26
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Wed Feb 21 11:13:56 2018 -0500

    Merge pull request #326 from jntullo/bz_1546375
    
    Set user when queueing VM actions
    (cherry picked from commit c6e8b4f9507292bf98032a107d67a55613b50243)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1565259
```

**Comment by bdunne** [comment]
This change depends on schema changes that can not be backported to the fine branch, marking as fine/no.
@miq-bot rm_label fine/yes
@miq-bot rm_label fine/conflict
@miq-bot add_label fine/no

**Pull request merged**